docs/en/04_advanced-configuration.md

```diff
@@ -0,0 +1,248 @@
+---
+title: Advanced Configuration
+summary: Further improving performance
+icon: sliders-h
+---
+
+# Advanced configuration
+
+As long as you respect the provided interfaces, any modifications will work well with the default engine. This means
+that the only thing left for you to do is to define behaviours:
+
+- what triggers publishing of what ([`StaticPublishingTrigger`](api:SilverStripe\StaticPublishQueue\Contract\StaticPublishingTrigger) interface)
+- what URLs belong to an object and need to be updated when object changes ([`StaticallyPublishable`](api:SilverStripe\StaticPublishQueue\Contract\StaticallyPublishable) interface).
+
+You can provide your own implementations by creating new extensions (see `PublishableSiteTree` for a simple
+example). You apply these as normal through the config system. You can also implement the interfaces directly on your
+classes without using extensions.
+
+> [!TIP]
+> When providing your own implementations you need to make sure you also cater for the default behaviour you might be
+> replacing. There is no "parent" you can call here because of how the extensions have been structured, and there is only
+> one "dominant" extension that will be executed.
+
+Some implementations of the aforementioned interfaces are bundled with this module. The [`PublishableSiteTree`](api:SilverStripe\StaticPublishQueue\Extension\Publishable\PublishableSiteTree) supports
+the basic static publishing functionality for SiteTree objects - updating itself and the parent on change, and
+deleting itself and updating parent on deletion.
+
+## Further improved performance
+
+Static Publish Queue has been built to balance (developer) usability (and featureset) with performance. As such
+there are ways to get better performance from the module if you're willing to accept some limitations / drawbacks.
+
+### Invoke the static request handler before the composer autoloader
+
+The composer autoloader allows us to include the `staticrequesthandler.php` file without knowledge of where it lives in
+the filesystem as we make use of the [`include-path` property](https://getcomposer.org/doc/04-schema.md#include-path).
+However, the composer autoloader can take up the majority of the processing time when serving a static page. As such you
+can tightly couple to the vendor path to provide a performance benefit.
+
+To do so, make your `index.php` file look like this (NB: if you're running using the `public` folder (SS 4.1+), you'll
```

Comment:
As of [5.0.0](https://docs.silverstripe.org/en/5/changelogs/5.0.0/):
> The use of the `public/` directory for the public web root is now mandatory.

Remove this `NB` note and update the paths as appropriate.

Reply:
Updated